x-pack/filebeat/tests/integration/framework_test.go

```diff
@@ -134,3 +134,34 @@ func (b *BeatProc) openLogFile() *os.File {
 
 	return f
 }
+
+// createTempDir creates a temporary directory that can persist
+// after the test finishes.
+//
+// If the tests are run with -v, the temporary directory will
+// be logged.
+//
+// Call the returned cleanup function to remove
+// the temporary directory. If there are any errors
+// cleanup will call t.Error
+func createTempDir(t *testing.T) (string, func()) {
+	tempDir, err := filepath.Abs(filepath.Join("../../build/integration-tests/",
+		fmt.Sprintf("%s-%d", t.Name(), time.Now().Unix())))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := os.MkdirAll(tempDir, 0766); err != nil {
+		t.Fatalf("cannot create tmp dir: %s, msg: %s", err, err.Error())
+	}
+	t.Logf("Temporary directory: %s", tempDir)
+
+	cleanup := func() {
```

Comment:
could we just add this to t.Cleanup, and have it check t.Failed() to see if it should delete the directory or not?

Reply:
I didn't know about `t.Failed()`, thanks a lot @leehinman! I'll implement it :D

Reply:
Done on 17c4ab90e1